src/open_inwoner/openzaak/tests/test_case_detail.py

```diff
@@ -2327,3 +2327,17 @@ def test_kcm_redirect(self, m):
             status_code=302,
             target_status_code=200,
         )
+
+    @patch.object(
+        ContactmomentenClient,
+        "retrieve_objectcontactmomenten_for_zaak",
+        autospec=True,
+        return_value=[None],
```

Comment:
This mock is not accurate and doesn't seem to test the modification. 

It makes `retrieve_objectcontactmomenten_for_zaak()` return a list with a single None value.

While the code change doesn't handle that. It loops over the `objectcontactmomenten` as if they can't be null and then does a null check on an attribute of an object in that list.

Reply:
The modification ensures sure that `object.contactmoment` is not `None`, so that `contactmoment.registratiedatum` further down doesn't raise `AttributeError`. `getattr(None, "contactmoment", None)` returns `None`, in exactly the same way that `getattr(objectcontactmoment, "contactmoment", None)` would return `None` for an `objectcontactmoment` that happens to lack a `contactmoment`. The test will therefore pass with the modification and fail without it. I didn't see a reason to create a fake `objectcontactmoment`; the mechnism is exactly the same (`getattr -> AttributeError -> None`).

Also, the code does not assume AFAIK that `objectcontactmomenten` can't be null; it creates an empty list, calls `retrieve_objectcontactmomenten_for_zaak` to populate it, then loops over the list.

Reply:
You made a modification that handles if an external sub resource is None, so the test should test with that same subresource being None. And not rely on some random other None value somewhere that happens to make the test pass for some unrelated reason. 

With this test we still don't know if you modification works or makes sense, because the situation it handles is not present in the test.

Reply:
You made a modification that handles if an external sub resource is None, so the test should test with that same subresource being None. And not rely on some random other None value somewhere that happens to make the test pass for some unrelated reason.

With this test we still don't know if you modification works or makes sense, because the situation it handles is not present in the test.